src/components/QuranReader/ReflectionView/ReflectionBodyContainer/index.tsx

```diff
@@ -36,6 +40,19 @@ const ReflectionBodyContainer = ({
   const { translationFontScale } = useSelector(selectQuranReaderStyles, shallowEqual);
   const { lang } = useTranslation();
 
+  const onReflectionViewed = useCallback((reflectionContainer: Element) => {
+    const postId = reflectionContainer.getAttribute('data-post-id');
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    postReflectionViews(postId).catch(() => {});
+  }, []);
```

Comment:
I think we need to have some sort of caching & batching here because:
1. If you scroll down and go up again, you'll send 2 requests (or more depending on how much you scroll) for the same reflection. 
2. As you scroll, you probably don't see reflections 1 by 1, so we can group reflections viewed at a time and send only 1 request.

Reply:
1. This is intentional. If you scroll past a reflection and stay long enough for the timeout to trigger then scrolled to it again it and stayed long enough for the timeout to trigger again, we should send two requests.
2. We don't fire the API call instantly but rather wait for a few seconds. If the user scrolls a bit making the post container not 100% visible we will clear the timeout so it will stop. I believe this makes it un-necessary to have queuing in place given.